osu.Game/Overlays/DirectOverlay.cs

```diff
@@ -298,13 +298,15 @@ private void updateSearch()
                 Task.Run(() =>
                 {
                     var onlineIds = response.Select(r => r.OnlineBeatmapSetID).ToList();
-                    var presentOnlineIds = beatmaps.QueryBeatmapSets(s => onlineIds.Contains(s.OnlineBeatmapSetID)).Select(r => r.OnlineBeatmapSetID).ToList();

```

Comment:
couldn't you just have added the pending check here rather than making even more filtering steps?

Reply:
There are not more filtering steps. I just split up one long condition on two lines.

I don't need to check the pending deletion for `presentOnlineIds` as they are already found there. The point is that they are removed later on which is why I removed the select and kept the beatmaps (we queried them anyways so there shouldn't be a downside to this).

The last line was split up as it was getting way too long. Can put them back together if you want though.

Reply:
Yikes I'mestupid. You meant checking there to NOT find them in the first place.

Reply:
splitting up a variable assignment in linq is generally only a good idea when it's used in a different way via that variable, not just chained to the next thing (which is why i assumed there were extra conditionals added). it's hard to mentally parse if you start doing this.